config/rhoai/prometheus_rule.yaml

```diff
@@ -0,0 +1,111 @@
+          description: "CPU load is below 20% for 1 day. Consider reducing the number of CPUs."
+      - alert:  HostHighCpuLoad
+        expr: (100 - (rate(node_cpu_seconds_total{mode="idle"}[30m]) * 100) > 80) * on(instance) group_left(nodename) node_uname_info{nodename=~".+"}
+        for: 1d
+        labels:
+          severity: info
+        annotations:
+          summary: Host high CPU load (instance {{ $labels.instance }})
+          description: "CPU load is above 80% for 1 day. Consider increasing the number of CPUs"
+      - alert: ContainerLowMemoryUsage 
+        expr: (sum(container_memory_working_set_bytes{name!=""}) BY (instance, name) / sum(container_spec_memory_limit_bytes > 0) BY (instance, name) * 100) < 20
+        for: 1d
+        labels:
+          severity: info
+        annotations:
+          summary: Container Low Memory Usage (instance {{ $labels.instance }})
+          description: Container memory usage is below 20% for 1 day. Consider reducing available memory.
+      - alert: ContainerHighMemoryUsage 
+        expr: (sum(container_memory_working_set_bytes{name!=""}) BY (instance, name) / sum(container_spec_memory_limit_bytes > 0) BY (instance, name) * 100) > 80
+        for: 1d
+        labels:
+          severity: info
+        annotations:
+          summary: Container High Memory Usage (instance {{ $labels.instance }})
+          description: Container memory usage is above 80% for 1 day. Consider increasing available memory.
+      - alert: LowClusterQueueResourceUsage
+        expr: sum(kueue_cluster_queue_resource_usage) by (cluster_queue) / sum(kueue_cluster_queue_nominal_quota) by (cluster_queue) < 0.2
+        for: 1d
+        labels:
+          severity: info
+        annotations:
+          summary: Low resource usage in cluster queue {{ $labels.cluster_queue }}
+          description: The resource usage in cluster queue {{ $labels.cluster_queue }} is below 20% of its nominal quota for more than 1 day.
+      - alert: HighAcceleratorUsage
+        expr: sum(kueue_cluster_queue_resource_usage{resource="nvidia.com/gpu"}) by (cluster_queue) / sum(kueue_cluster_queue_nominal_quota{resource="nvidia.com/gpu"}) by (cluster_queue) > 0.9
+        for: 1d
+        labels:
+          severity: info
+        annotations:
+          summary: High Accelerator Usage.
+          description: Accelerator usage is above 90% of the cluster queue quota for 1 day. Consider increasing available accelerators
+      - alert: LowAcceleratorUsage
+        expr: sum(kueue_cluster_queue_resource_usage{resource="nvidia.com/gpu"}) by (cluster_queue) / sum(kueue_cluster_queue_nominal_quota{resource="nvidia.com/gpu"}) by (cluster_queue) < 0.2
+        for: 1d
+        labels:
+          severity: info
+        annotations:
+          summary: Low Accelerator Usage.
+          description: Accelerator usage is below 20% of the cluster queue quota for 1 day. Consider reducing available accelerators
```

Comment:
Is this possible to  group by `resource` so that if an admin adds a new accelerator we don't know about they'll automatically get alerts about high and low accelerator usage for it?

Reply:
Yes they can be grouped by resource so then we could have just 1 alert covering all resources instead of separating it our by resource like this: 
```
- alert: LowClusterQueueResourceUsage
        expr: sum(kueue_cluster_queue_resource_usage) by (cluster_queue, resource) / sum(kueue_cluster_queue_nominal_quota) by (cluster_queue, resource) < 0.2
        for: 1d
        labels:
          severity: info
        annotations:
          summary: Low {{ $labels.resource }} resource usage in cluster queue {{ $labels.cluster_queue }}
          description: The {{ $labels.resource }} resource usage in cluster queue {{ $labels.cluster_queue }} is below 20% of its nominal quota for more than 1 day.
```
```
- alert: HighClusterQueueResourceUsage
        expr: sum(kueue_cluster_queue_resource_usage) by (cluster_queue, resource) / sum(kueue_cluster_queue_nominal_quota) by (cluster_queue, resource) > 0.8
        for: 1d
        labels:
          severity: info
        annotations:
          summary: High {{ $labels.resource }}  resource usage in cluster queue {{ $labels.cluster_queue }}
          description: The {{ $labels.resource }} resource usage in cluster queue {{ $labels.cluster_queue }} is higher than 80% of its nominal quota for more than 1 day.
```
I spoke to @astefanutti about this previously and he didn't see any value in including high usage as it would be expected, but to include something around requests for resources being way higher than quota (which I have included in another alert).
wdyt?

Reply:
I've consolidated these into 1 alert by resource. Thanks.